### DISCUSSION TITLE
Battery Inverter/Ambient temps?

### DISCUSSION THREAD

**Discussion by kme77** [StealthChesnut/HA-FoxESS-Modbus#69] [Q&A]
Just a quick question. Due to the cold weather I've been paying more attention to battery temps and noticed that the 'Ambient' and 'Inv' sensor temps appear to be inverted in the integration compared to the Fox cloud app. Is this an error or am I missing something?

Cheers. 👍

**Comment by spmaid** [comment]
> Just a quick question. Due to the cold weather I've been paying more attention to battery temps and noticed that the 'Ambient' and 'Inv' sensor temps appear to be inverted in the integration compared to the Fox cloud app. Is this an error or am I missing something?
> 
> Cheers. 👍

I wonder what the ambient temp is, as mine shows 30 at the moment?  It aint 30 out in my garage that's for sure! :-)

**Reply by kme77** [reply]
I was wondering the same, mine is showing 20C and it's FREEZING in my garage! 😂

**Reply by spmaid** [reply]
I wonder if it's in the air space inside the Inverter, and the Inverter temp is directly on the inverter. 

I actually installed some CPU fans to the bottom of my inverter and it helped massively in summer to keep the temp down.  Turned them off for now though!

**Reply by kme77** [reply]
It would be nice to get a definitive answer from fox as to where all the temps are being measured from.

**Comment by tagware** [comment]
In F' rather than C?

**Reply by spmaid** [reply]
I'd say no, it's not that cold either...

**Reply by kme77** [reply]
Both the cloud and integration are using Celcius 👍
And the readings are literally swapped as if they're addressing the others Modbus address.

**Comment by Daibutt** [comment]
![image](https://user-images.githubusercontent.com/69276073/208242741-c9eb1f3a-ecc2-47b0-930e-883bf57d5045.png)

Wiki for this integration, this is what the data register reference says........
Edit: Just realised LAN modbus addresses may be different. 
Alos judging by descriptions it sort of makes sense for mine; ambient 29C (CPU), Inverter 20C (cooling fins), my loft temp is  currently 9C.

**Reply by spmaid** [reply]
Agreed, this does make sense, even if the FoxESS descriptions seem a bit off :)
